nn_unet_scripts/convert_from_bids_to_nnunetv2_format.py

```diff
@@ -0,0 +1,322 @@
+        json.dump(data, f, indent=2)
+
+
+def process_images(
+    datapath: Path,
+    out_folder: str,
+    particpant_to_sample_dict: Dict[str, List[str]],
+    bids_to_nnunet_dict: Dict[str, int],
+    dataset_name: str,
+    is_test: bool = False,
+):
+    """
+    Processes all image files in each subject's directory.
+
+    Parameters
+    ----------
+    datapath : Path
+        Path to the data directory.
+    out_folder : str
+        Output directory to save processed images.
+    particpant_to_sample_dict : Dict[str, List[str]]
+        Dictionary mapping participant IDs to sample IDs.
+    bids_to_nnunet_dict : Dict[str, int]
+        Dictionary mapping subject names to case IDs.
+    dataset_name : str
+        Name of the dataset.
+    is_test : bool, optional
+        Boolean flag indicating if the images are for testing, by default False.
+    """
+    folder_type = "imagesTs" if is_test else "imagesTr"
+    image_suffix = "_0000"
+
+    for subject in particpant_to_sample_dict.keys():
+        for image in particpant_to_sample_dict[subject]:
+            case_id = bids_to_nnunet_dict[str((subject, image))]
+            image_path = os.path.join(
+                datapath, subject, "micr", f"{subject}_{image}_SEM.png"
+            )
+            img = cv2.imread(str(image_path), cv2.IMREAD_GRAYSCALE)
+            fname = f"{dataset_name}_{case_id:03d}{image_suffix}.png"
+            cv2.imwrite(os.path.join(out_folder, folder_type, fname), img)
+
+
+def process_labels(
+    datapath: Path,
+    out_folder: str,
+    particpant_to_sample_dict: Dict[str, List[str]],
+    bids_to_nnunet_dict: Dict[str, int],
+    dataset_name: str,
+    label_type: Literal["axonmyelin", "myelin", "axon"] = "axonmyelin",
+):
+    """
+    Processes label images from a list of subjects, matching each image with the label having the largest 'N' number.
+
+    Parameters
+    ----------
+    datapath : Path
+        Path to the data directory.
+    out_folder : str
+        Output directory to save processed label images.
+    particpant_to_sample_dict : Dict[str, List[str]]
+        Dictionary mapping participant IDs to sample IDs.
+    bids_to_nnunet_dict : Dict[str, int]
+        Dictionary mapping subject names to case IDs.
+    dataset_name : str
+        Name of the dataset.
+    """
+    for subject in particpant_to_sample_dict.keys():
+        for image in particpant_to_sample_dict[subject]:
+            case_id = bids_to_nnunet_dict[str((subject, image))]
+            label_path = os.path.join(
+                datapath,
+                "derivatives",
+                "labels",
+                subject,
+                "micr",
+                f"{subject}_{image}_SEM_seg-{label_type}-manual.png",
+            )
+            label = cv2.imread(str(label_path), cv2.IMREAD_GRAYSCALE)
+            mapping_dict = {0: 0, 127: 1, 128: 1, 130: 1, 233: 2, 255: 2}
+            label = np.vectorize(mapping_dict.get)(label)
```

Comment:
```suggestion
            label = label / 127
```
This might be safer for future datasets. The 128, 130 and 233 values in the masks are mistakes, but this often happens so if you add images that have different glitched values the mapping would fail.

Reply:
Ahh I understand why you did it this way... so that the function could also process axon and myelin labels, depending on the `label_type` argument.

The problem here is that if you have `label_type=myelin`, for example, on line 137 you will read the myelin label. Its values are 0 and 255 because it is a binary mask. Your mapping function will assign the myelin to the second class, which is a different behavior than when you feed the axonmyelin masks.

Reply:
I suggest that this function should only read the axonmyelin mask on line 137. Based on the `label_type`, your function can map to the right value.

Reply:
Yes you're right good catch! I will change it.